src/ast/passes/resource_analyser.cpp

```diff
@@ -289,6 +297,60 @@ void ResourceAnalyser::visit(Call &call)
       resources_.str_buffers++;
   }
 
+  /* Aggregation functions like count/sum/max are always called like:
+   *   @ = count()
+   * Thus, we visit AssignMapStatement AST node which visits the map and
+   * assigns a map key buffer. Thus, there is no need to assign another
+   * buffer here.
+   *
+   * The exceptions are:
+   * 1. lhist/hist because the map key buffer includes both the key itself
+   *    and a function pointer to the linear/log2 functions.
```

Comment:
Not sure it's a function pointer. Looks like the result of a call to log2 - ie the bucket. right?

Reply:
Ah I think you're right, I misread the code, I will fix